File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.java
class Solution {
    public boolean winnerOfGame(String colors) {
        int numA = 0;
        int numB = 0;
        int colorsSize = colors.length();
        for (int i = 1; i < colorsSize - 1; ++i) {
            Character curr = colors.charAt(i);
            Character prev = colors.charAt(i - 1);
            Character next = colors.charAt(i + 1);
            if (curr == prev && curr == next) {
                if (curr == 'A') {
                    numA += 1;
                } else {
                    numB += 1;
                }
            }
        }
        return numA > numB;
    }
}

File: 2038-remove-colored-pieces-if-both-neighbors-are-the-same-color/2038-remove-colored-pieces-if-both-neighbors-are-the-same-color.py
class Solution:
    def winnerOfGame(self, colors: str) -> bool:
        N = len(colors)
        A, B = 0, 0
        for i in range(1, N - 1):
            if colors[i] == colors[i - 1] == colors[i + 1]:
                if colors[i] == "A":
                    A += 1
                else:
                    B += 1
        return A > B
